specs/WebResourceResponseReceived.md

```diff
@@ -0,0 +1,232 @@
+                    [this, webResourceRequest, webResourceResponse](HRESULT result) {
+                        // The response might not have a body.
+                        bool populatedBody = SUCCEEDED(result);
+                        
+                        std::wstring message =
+                            L"{ \"kind\": \"event\", \"name\": "
+                            L"\"WebResourceResponseReceived\", \"args\": {"
+                            L"\"request\": " +
+                            RequestToJsonString(webResourceRequest.get()) +
+                            L", "
+                            L"\"response\": " +
+                            ResponseToJsonString(webResourceResponse.get()) + L"}";
+
+                        message +=
+                            WebViewPropertiesToJsonString(m_webview.get());
+                        message += L"}";
+                        PostEventMessage(message);
+                        return S_OK;
+                    })
+                    .Get());
+
+            return S_OK;
+        })
+        .Get(),
+    &m_webResourceResponseReceivedToken);
+```
+
+## C#
+```c#
+WebView.WebResourceResponseReceived += WebView_WebResourceResponseReceived;
+
+// Note: modifications made to request and response are ignored
+private async void WebView_WebResourceResponseReceived(object sender, CoreWebView2WebResourceResponseReceivedEventArgs e)
+{
+    // Actual headers sent with request
+    foreach (var current in e.Request.Headers)
+        Console.WriteLine(current);
+
+    // Headers in response received
+    foreach (var current in e.Response.Headers)
+        Console.WriteLine(current);
+
+    // Status code from response received
+    int status = e.Response.StatusCode;
+    if (status == 200)
+    {
+        // Handle
+        Console.WriteLine("Request succeeded!");
+
+        // Get response body
+        try
+        {
+            await e.PopulateResponseContentAsync();
+            DoSomethingWithResponseBody(e.Response.Content);
+        }
+        catch (COMException ex)
+        {
+            // A COMException will be thrown if the request has no body.
+        }
+    }
+}
+```
+
+
+# Remarks
+Calling `PopulateResponseContent` will fail/throw a COMException if the response
+has no body. Note the body for redirect responses is ignored.
+
+
+# API Notes
+See [API Details](#api-details) section below for API reference.
+
+
+# API Details
+## COM
+```cpp
+library WebView2
+{
+    interface ICoreWebView2 : IUnknown
+    {
+        // ...
+
+        /// Add an event handler for the WebResourceResponseReceived event.
+        /// WebResourceResponseReceived event is raised after the WebView has received
+        /// and processed the response for a WebResource request. The event args
+        /// include the WebResourceRequest as committed and the WebResourceResponse
+        /// received, including any additional headers added by the network stack that
```

Comment:
Overpromises. Doesn't mention that Content is missing from the WebResourceResponse.

```cpp
/// not including the Content, but including any additional headers...
```